### DISCUSSION TITLE
Erro Select Option Programatically - DropDown

### DISCUSSION THREAD

**Discussion by blockchain-dei** [primefaces/community#116] [PrimeVue]
Hi,

I've been using primeVue, but I faced a problem with the dropdown component.

I have a drop-down filled with options but when I try to select one option programmatically, my program enters a looping (very strange).

I'm selecting the option, changing the v-model variable. Is it the correct form ?

Ex:
optOptions: [
                { name: 'OP1', value: 'OP1' },
                { name: 'OP2', value: 'OP2' },
                { name: 'OP3', value: 'OP3' },

            ],

    <Dropdown v-model="selectedOption      " :options="optOptions"  optionLabel="name"  />
 methods: {
        FunctionChangeOption() {
        console.log(loop).
          this.selectedOption      =  {'Option' ....}

}

**Comment by tugcekucukoglu** [comment]
Hi,

As far as I can see `options` collection is an array of objects whose keys are name and value. But in the method, the v-model is changed to a different type of object. You can try this: 

```
methods: {
     FunctionChangeOption() {
           console.log(loop).
           this.selectedOption = { name: 'OP3', value: 'OP3' }
    }
}
```

**Comment by blockchain-dei** [comment]
> name: 'OP1', value: 'OP1'

Thank you Tug~ I appreciate your help.

But I'm doing exactly what you are saying. I'm changing the v-model, using a JsonObject, and it works (it selects the option programmatically !). But the problem is that strange behavior happens in FunctionChangeOption. It is called in-looping because changing the model fires some kind of event (changing v-model fire drop-down event, reload the page, in reload it change selected option that fire again the event that fires reload (looping).


Att,
Fernando.

**Comment by tugcekucukoglu** [comment]
Is there any reproducer link that simply shows the error?
